### PR TITLE
Refactor DatastoreRepository

### PR DESCRIPTION
**Pull request by MagoDelBlocco**
Added multiple code style changes:

* replaced Objects.requireNonNull with Preconditions.checkNotNull
* replaced import Preconditions with import static Preconditions.checkArgument & Preconditions.checkNotNull
* added `@Override` annotation for index related methods
* appended the `@throws` block to the inherited JDoc for all applicable methods

### CODE REVIEW

**Review by holounic — APPROVED** [review]
LGTM

File: src/main/java/com/google/graphgeckos/dashboard/storage/DatastoreRepository.java [hunk 1]
```diff
@@ -124,11 +131,11 @@ public boolean updateRevisionEntry(@NonNull BuildBotData updateData) {
   /**
    * {@inheritDoc}
    *
-   * @throws NullPointerException if the {@code commitHash} is null.
+   * @throws NullPointerException if {@code commitHash} is null.
    */
   @Override
   public boolean deleteRevisionEntry(@NonNull String commitHash) {
-    Objects.requireNonNull(commitHash);
+    checkNotNull(commitHash, "commitHash cannot be null");
```
Comment (krasimir-google) [inline src/main/java/com/google/graphgeckos/dashboard/storage/DatastoreRepository.java:138]: nit: I think the message is unnecessary

File: src/main/java/com/google/graphgeckos/dashboard/storage/DatastoreRepository.java [hunk 2]
```diff
@@ -171,17 +178,24 @@ public boolean deleteRevisionEntry(@NonNull String commitHash) {
   /**
    * {@inheritDoc}
    *
-   * @throws NullPointerException if the {@code commitHash} is null.
+   * @throws NullPointerException if the {@code commitHash} is null
    */
   @Override
   public BuildInfo getRevisionEntry(@NonNull String commitHash) {
-    Objects.requireNonNull(commitHash);
+    checkNotNull(commitHash, "commitHash should not be null");
```
Comment (krasimir-google) [inline src/main/java/com/google/graphgeckos/dashboard/storage/DatastoreRepository.java:185]: I think the message is unnecessary

**Review by krasimir-google — APPROVED** [review]
looks good with 2 nits

File: src/main/java/com/google/graphgeckos/dashboard/storage/DatastoreRepository.java (hunk 2 above)
Comment (MagoDelBlocco) [reply src/main/java/com/google/graphgeckos/dashboard/storage/DatastoreRepository.java:185]: I'll remove them before merging. Thank you!

**Pull request merged**